orbstation/quirks/plurality.dm

```diff
@@ -0,0 +1,315 @@
+/obj/item/plural_system_chip/proc/update_system_name(new_system_name)
+	system_name = new_system_name
+	name = "[initial(name)] {[system_name]}"
+
+/obj/item/plural_system_chip/attackby(obj/item/W, mob/user, params)
+	if(W.GetID())
+		if(allowed(user))
+			locked = !locked
+			system_name_locked = FALSE
+			balloon_alert(user, locked ? "locked" : "unlocked")
+			update_appearance()
+			return
+		else
+			balloon_alert(user, "access denied!")
+			return
+
+// Code that allows ID cards to have plural system chips attached to them and interface with them to change the label of the ID.
+
+/obj/item/card/id
+	/// Attached plural system chip, if any.
+	var/obj/item/plural_system_chip/plural_system
+	/// Whether a plural system chip can be added to the ID.
+	var/plural_system_compatible = TRUE
+
+/obj/item/card/id/attack_self(mob/user)
+	if(user.incapacitated())
+		return
+	if(!user.is_holding(src))
+		return
+	// if there is a plural system chip attached and the system has members, shows a popup through which
+	if(plural_system?.system_member_names.len && plural_system_compatible)
+		var/list/menu_options = plural_system.system_member_names.Copy()
+		menu_options |= "Show ID"
+		var/popup_input = tgui_input_list(user, "Select the current fronter:", "Plural System Interface", menu_options)
+		if(!popup_input)
+			return
+		if(popup_input != "Show ID")
+			plural_system.fronter_name = popup_input
+			balloon_alert(user, "[popup_input]")
+			to_chat(user, span_notice("You set [plural_system]'s current fronter to [plural_system.fronter_name]."))
+			update_label()
+			return
+	..() // if there is no plural system chip or the "Show ID" option was chosen, just show the ID like normal
+
+/obj/item/card/id/attackby(obj/item/item, mob/user, params)
+	if(istype(item, /obj/item/plural_system_chip))
+		if(!plural_system_compatible)
+			to_chat(user, span_warning("This ID card is incompatible with plural system chips."))
+			return
+
+		var/obj/item/plural_system_chip/new_chip = item
+
+		if(plural_system)
+			to_chat(user, span_warning("There's already a plural system chip installed. Use a screwdriver to remove it."))
+			return
+		if(!registered_name)
+			to_chat(user, span_warning("ERROR: [src] has no registered name. Take it to an identification console."))
+			return
+		if(!new_chip.system_member_names.len)
+			to_chat(user, span_warning("[new_chip] has no system members registered with it."))
+			return
+		if(!user.transferItemToLoc(new_chip, src))
+			return
+
+		plural_system = new_chip
+		to_chat(user, span_notice("You insert [plural_system] into [src]."))
+		plural_system.fronter_name = plural_system.system_member_names[1]
+		plural_system.locked = TRUE
+		plural_system.update_appearance()
+		update_label()
+		playsound(src, 'sound/machines/pda_button1.ogg', 50, TRUE)
+
+/obj/item/card/id/screwdriver_act(mob/living/user, obj/item/screwtool)
+	if(!plural_system)
+		to_chat(user, span_warning("There's no plural system chip installed."))
```

Comment:
I think this _may_ not need a specific chat warning, since an ID with a chip on it will have it visible - there won't be many cases where someone is specifically trying to remove a chip that isn't there, so the warning won't be helpful to many people.

Not that I'm sure why anyone would be using a screwdriver on an ID otherwise. This is another suggestion that isn't very important.

Reply:
I don't think it's really necessary to remove this - it's helpful for players to get feedback on why something isn't working if they're trying it.

Reply:
Fair enough!